llvm/lib/Transforms/Vectorize/VPlanUtils.cpp

```diff
@@ -66,6 +66,17 @@ bool vputils::isHeaderMask(const VPValue *V, const VPlan &Plan) {
                       m_One(), m_Specific(&Plan.getVF()))) ||
             IsWideCanonicalIV(A));
 
+  // For scalar plans, the header mask uses the scalar steps.
+  if (match(V,
+            m_ICmp(m_ScalarIVSteps(
```

Comment:
Looks like this pattern is now used twice in this function, i.e.

```
m_ScalarIVSteps(m_Specific(Plan.getVectorLoopRegion()->getCanonicalIV()),
                m_One(), m_Specific(&Plan.getVF()))
```

Is it worth creating a specific pattern match for this? Something like

```
  m_CanonicalScalarIVSteps(Plan)
```

?